romancal/assign_wcs/pointing.py

```diff
@@ -45,3 +50,66 @@ def v23tosky(input_model, wrap_v2_at=180, wrap_lon_at=360):
     )
     model.name = "v23tosky"
     return model
+
+
+def dva_corr_model(va_scale, v2_ref, v3_ref):
+    """
+    Create transformation that accounts for differential velocity aberration
+    (scale).
+
+    Parameters
+    ----------
+    va_scale : float, None
+        Ratio of the apparent plate scale to the true plate scale. When
+        ``va_scale`` is `None`, it is assumed to be identical to ``1`` and
+        an ``astropy.modeling.models.Identity`` model will be returned.
+
+    v2_ref : float, None
+        Telescope ``v2`` coordinate of the reference point in ``arcsec``. When
+        ``v2_ref`` is `None`, it is assumed to be identical to ``0``.
+
+    v3_ref : float, None
+        Telescope ``v3`` coordinate of the reference point in ``arcsec``. When
+        ``v3_ref`` is `None`, it is assumed to be identical to ``0``.
+
+    Returns
+    -------
+    va_corr : astropy.modeling.CompoundModel, astropy.modeling.models.Identity
+        A 2D compound model that corrects DVA. If ``va_scale`` is `None` or 1
+        then `astropy.modeling.models.Identity` will be returned.
+
+    """
+    if va_scale is None or va_scale == 1:
+        return Identity(2)
+
+    if va_scale <= 0:
+        log.warning("Given velocity aberration scale %s", va_scale)
+        log.warning(
+            "Velocity aberration scale must be a positive number. Setting to 1.0"
+        )
+        va_scale = 1.0
+
+    va_corr = Scale(va_scale, name="dva_scale_v2") & Scale(
+        va_scale, name="dva_scale_v3"
+    )
+
+    if v2_ref is None:
+        v2_ref = 0
+
+    if v3_ref is None:
+        v3_ref = 0
+
+    if v2_ref == 0 and v3_ref == 0:
+        return va_corr
+
+    # NOTE: it is assumed that v2, v3 angles and va scale are small enough
+    # so that for expected scale factors the issue of angle wrapping
+    # (180 degrees) can be neglected.
+    v2_shift = (1 - va_scale) * v2_ref
+    v3_shift = (1 - va_scale) * v3_ref
```

Comment:
This looks good to me.  FWIW, I went down this rabbit hole again
https://github.com/spacetelescope/rad/issues/437
which permanently confuses me.  I don't think we want any changes here related to that, but you could confirm whether I'm confused.

Reply:
I was just confused.  The key point is that dva_corr_model(v2_ref, v3_ref) == (v2_ref, v3_ref), while the shift was telling me that that wasn't true.  But the shift exactly cancels out the motion of the center due to the scale and all is well.

Reply:
As per slack note, this has resolved.